src/core/Group.cpp

```diff
@@ -728,10 +710,11 @@ QSet<QUuid> Group::customIconsRecursive() const
 
 Group* Group::findGroupByUuid(const QUuid& uuid)
 {
-    Q_ASSERT(!uuid.isNull());
-    for (Group* group : groupsRecursive(true)) {
-        if (group->uuid() == uuid) {
-            return group;
+    if (!uuid.isNull()) {
```

Comment:
See above.

Reply:
same comment, changing this will create three return statements unnecessarily